### DISCUSSION TITLE
Teufel Raumfeld Integration v0.1.12-alpha4 on Home Assistant 2023.5.x

### DISCUSSION THREAD

**Discussion by 4711Raven** [B5r1oJ0A9G/teufel_raumfeld#46] [Q&A]
Hello,
I have been using the integration for my 5 Raumfeld  appliances for some time and have been very happy with it. Good Job Marc.
However, when I upgraded to version 2023.5, all the entities associated with the integration disappeared. After a rollback to version 2023.4 the Group and Room entities are working well again. The used database is MariaDB, Version: 2.6.1
I also have noticed that, for example, the 'Powerstate' entities are also no longer present. But this  depends not on Version 2023.5 because on 2023.4.  
Does anyone have an idea why this no longer works.

**Comment by 4711Raven** [comment]
In the log I found this:
You are using teufel_raumfeld, which is still in alpha phase and therefore subject to change. This includes, among other things, the addition, redesign or removal of functionality. Does this cause the issue?

**Reply by DonaldGermany** [reply]
Hi, 
I have the same problem, found this recommendation https://github.com/B5r1oJ0A9G/teufel_raumfeld/issues/44,
now Groups and Rooms are back, but they don't show the current status (play, pause, title, etc). Only when I use the HA mediaplayer to start the music it shows the status.

**Comment by 4711Raven** [comment]
following recommendation #44 I commented out line 133 
      entity_registry = await hass.helpers.entity_registry.async_get_registry() 
and then entered  line 
     entity_registry = hass.helpers.entity_registry.async_get(hass)

As DonaldGeramny wrote the Rooms are back and onyl play icon is shown but only if the device is Off or in Eco mode. If the device is Off state it must be switched to On or to Eco mode first before typing the play icon has the expected effect.
I did a test with a script that I had in place for an alarm clock automation where I switched the device into eco mode:

service: teufel_raumfeld.set_power_state
data:
  power_state: eco
  entity_id: sensor.room_bad_powerstate

**Comment by B5r1oJ0A9G** [comment]
Hi! As @DonaldGermany rightfully mentioned this was a bug or rather incompatibility to changes made in Home Assistant. It is fixed with release of [v0.1.12-alpha5](https://github.com/B5r1oJ0A9G/teufel_raumfeld/releases/tag/v0.1.12-alpha5)
